flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java

```diff
@@ -573,6 +574,20 @@ public static int getRunningContainers() {
 		return count;
 	}
 
+	public void waitUntilApplicationFinished(ApplicationId applicationId, int timeout) throws Exception {
+		Deadline deadline = Deadline.now().plus(Duration.ofSeconds(timeout));
+		while (deadline.hasTimeLeft()) {
+			YarnApplicationState state = yarnClient.getApplicationReport(applicationId).getYarnApplicationState();
+			if (state == YarnApplicationState.FINISHED ||
+					state == YarnApplicationState.FAILED ||
+					state == YarnApplicationState.KILLED) {
+				break;
+			} else {
+				sleep(500);
```

Comment:
also a private static final constant_milli
I think 100 ms is actually enough to wait in most of normal cases.

Reply:
OK